### ISSUE TITLE
Loki Query Builder: add label suggestions to label filter expression

### ISSUE THREAD

**Issue by matyax** [grafana/grafana#59216]
# Context

In the Query Builder, we have a filter for label values, which is very basic.

![imagen](https://user-images.githubusercontent.com/1069378/203558740-de12bbbf-ae4e-43a3-89b3-3b0f5382b53a.png)

# Expectations

- Change the text input for a select (with custom entry support) which displays actual labels. 
- In order to get these labels, you need to run the log query portion of the current query.

**Comment by ThinhNguyenHoang1** [comment]
Hello, can you assign me. I want to try implement this

**Comment by matyax** [comment]
Hey @ThinhNguyenHoang1 ! No need for assingments. You can just go ahead and send a PR our way.

Keep in mind that this is not a straightforward task, so you will need a bit of knowledge about Loki, the LokiDatasource class https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/datasource.ts and the Loki language provider class https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/LanguageProvider.ts

Good luck!

**Comment by ThinhNguyenHoang1** [comment]
Hey @matyax, after reading through the docs and playing around with the code. I found that we can run build query label filters with both the original and extracted label fields. 

But the retrieved label values from the dataSource.languageProvider includes only the original common labels. 

Do we expect the select here to includes the extracted labels as well ?

**Comment by matyax** [comment]
Yes @ThinhNguyenHoang1 , indexed, parsed, and structured metadata.
